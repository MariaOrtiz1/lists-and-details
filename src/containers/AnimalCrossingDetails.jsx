import React, { useState, useEffect } from 'react';
import { useParams } from 'react-router-dom';
import { findOneVillager } from '../services/animalCrossing';

const AnimalCrossingDetail = () => {
  const { id } = useParams();
  const [loading, setLoading] = useState(true);
  const [villager, setVillager] = useState({});

  useEffect(() => {
    findOneVillager(id)
      .then((villager) => setVillager(villager))
      .finally(() => setLoading(false));
  }, [id]);

  if(loading) return <img 
    src="https://i.giphy.com/media/XE7hLtlphwAK70mP3i/giphy.webp" 
    alt="loading spinner of flying package with balloon" 
    style={{
      'display': 'flex',
      'justifyContent': 'center',
      'alignSelf': 'center',
    }}/>;

  return (
    <div role="div" arai-label="villager">
      <p>{villager.id}</p>
      <p>{villager.name}</p>
      <img src={villager.image} alt={villager.name} style={{
        'height': '300px',
        'width': '250px',
        'backgroundColor': '#edcadb',
        'borderRadius': '30px',
        'border': '3px dotted #a27cbf',
      }}/>
      <p>{villager.species}</p>
      <p>{villager.personality}</p>
      <p>{villager.quote}</p>
    </div>
  );
};

export default AnimalCrossingDetail;
