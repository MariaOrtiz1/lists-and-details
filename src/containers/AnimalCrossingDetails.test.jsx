/* eslint-disable max-len */
import React from 'react';
import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router';
import { rest } from 'msw';
import { setupServer } from 'msw/node';
import AnimalCrossingDetails from './AnimalCrossingDetails';
import DetailData from '../data/detailData.json';

const mockServer = setupServer(
  rest.get('https://ac-vill.herokuapp.com/villagers/:id', (req, res, ctx) => {
    return res(ctx.json(DetailData));
  })
);

describe('AnimalCrossingDetails', () => {
  beforeAll(() => mockServer.listen());
  afterAll(() => mockServer.close());

  it('displays info about a single villager', async () => {
    const  { container }  = render(
      <MemoryRouter initialEntries={['/5f5fb4bbbfd05c2aed82e46e']}>
        <AnimalCrossingDetails />
      </MemoryRouter>
    );

    screen.getByAltText('loading spinner of flying package with balloon');

    await screen.findByText('Annalise', { exact: false });
    expect(container).toMatchSnapshot();
  });
});
