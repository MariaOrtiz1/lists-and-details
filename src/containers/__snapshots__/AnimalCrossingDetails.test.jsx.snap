// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnimalCrossingDetails displays info about a single villager 1`] = `
<div>
  <div
    arai-label="villager"
    role="div"
  >
    <p>
      5f5fb4bbbfd05c2aed82e46e
    </p>
    <p>
      Annalise
    </p>
    <img
      alt="Annalise"
      src="https://vignette.wikia.nocookie.net/animalcrossing/images/c/c4/Annalise_NH.png/revision/latest?cb=20200802153943"
      style="height: 300px; width: 250px; background-color: rgb(237, 202, 219); border-radius: 30px; border: 3px dotted #a27cbf;"
    />
    <p>
      horse
    </p>
    <p>
      snooty
    </p>
    <p>
      Let gossip in one ear and out the other.
    </p>
  </div>
</div>
`;
